packages/cli/src/executionLifecycleHooks/restoreBinaryDataId.ts

```diff
@@ -32,28 +33,40 @@ export async function restoreBinaryDataId(
 		return;
 	}
 
-	const { runData } = run.data.resultData;
+	try {
+		const { runData } = run.data.resultData;
 
-	const promises = Object.keys(runData).map(async (nodeName) => {
-		const binaryDataId = runData[nodeName]?.[0]?.data?.main?.[0]?.[0]?.binary?.data?.id;
+		const promises = Object.keys(runData).map(async (nodeName) => {
+			const binaryDataId = runData[nodeName]?.[0]?.data?.main?.[0]?.[0]?.binary?.data?.id;
```

Comment:
I think it would make more sense to have the try catch inside the map function, as the tasks are independent of each others and any one or multiple of them could in theory throw. Otherwise if one of them throws the function will return even if the other promises are still going. Might as well extract it to its own function while at it.

Other option would be to use [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and check the result afterwards and log possible errors

Reply:
TBH @tomi I did implement the change but given we have to then update the binary data id after the renaming process happens, it turned out so ugly by having to use `then` as well as `allSettled` in a `.forEach` pushing things to a promise array that I was weighing the pros and cons.

If we fail to rename some files even though this is independent, the impact is that you might end up with some files not being deleted when the execution is removed.

This is not great, but this also only happens if there's something else went wrong with the execution, where the binary file was not written to disk in the first place, so this is an edge case and the impact is minimal, so I  went for code legibility.